synapse/util/caches/ttlcache.py

```diff
@@ -159,12 +159,12 @@ def expire(self) -> None:
             del self._expiry_list[0]
 
 
-@attr.s(frozen=True, slots=True)
-class _CacheEntry:
```

Comment:
CI is unhappy here.

<details>

```
2021-09-27T15:54:47.4244222Z [ERROR]
2021-09-27T15:54:47.4244683Z Traceback (most recent call last):
2021-09-27T15:54:47.4251489Z   File "/home/runner/work/synapse/synapse/.tox/py/lib/python3.6/site-packages/twisted/trial/runner.py", line 693, in loadByName
2021-09-27T15:54:47.4252855Z     return self.suiteFactory([self.findByName(name, recurse=recurse)])
2021-09-27T15:54:47.4254696Z   File "/home/runner/work/synapse/synapse/.tox/py/lib/python3.6/site-packages/twisted/trial/runner.py", line 453, in findByName
2021-09-27T15:54:47.4255759Z     obj = reflect.namedModule(searchName)
2021-09-27T15:54:47.4257137Z   File "/home/runner/work/synapse/synapse/.tox/py/lib/python3.6/site-packages/twisted/python/reflect.py", line 157, in namedModule
2021-09-27T15:54:47.4258112Z     topLevel = __import__(name)
2021-09-27T15:54:47.4258979Z   File "/home/runner/work/synapse/synapse/tests/rest/client/test_third_party_rules.py", line 27, in <module>
2021-09-27T15:54:47.4259789Z     from tests import unittest
2021-09-27T15:54:47.4260567Z   File "/home/runner/work/synapse/synapse/tests/unittest.py", line 38, in <module>
2021-09-27T15:54:47.4261634Z     from synapse.federation.transport import server as federation_server
2021-09-27T15:54:47.4262824Z   File "/home/runner/work/synapse/synapse/synapse/federation/transport/server/__init__.py", line 21, in <module>
2021-09-27T15:54:47.4264029Z     from synapse.federation.transport.server._base import (
2021-09-27T15:54:47.4265283Z   File "/home/runner/work/synapse/synapse/synapse/federation/transport/server/_base.py", line 31, in <module>
2021-09-27T15:54:47.4266216Z     from synapse.server import HomeServer
2021-09-27T15:54:47.4267084Z   File "/home/runner/work/synapse/synapse/synapse/server.py", line 54, in <module>
2021-09-27T15:54:47.4268186Z     from synapse.federation.federation_client import FederationClient
2021-09-27T15:54:47.4275541Z   File "/home/runner/work/synapse/synapse/synapse/federation/federation_client.py", line 58, in <module>
2021-09-27T15:54:47.4277013Z     from synapse.federation.transport.client import SendJoinResponse
2021-09-27T15:54:47.4278483Z   File "/home/runner/work/synapse/synapse/synapse/federation/transport/client.py", line 33, in <module>
2021-09-27T15:54:47.4280832Z     from synapse.http.matrixfederationclient import ByteParser
2021-09-27T15:54:47.4282277Z   File "/home/runner/work/synapse/synapse/synapse/http/matrixfederationclient.py", line 67, in <module>
2021-09-27T15:54:47.4283684Z     from synapse.http.federation.matrix_federation_agent import MatrixFederationAgent
2021-09-27T15:54:47.4285068Z   File "/home/runner/work/synapse/synapse/synapse/http/federation/matrix_federation_agent.py", line 41, in <module>
2021-09-27T15:54:47.4286572Z     from synapse.http.federation.well_known_resolver import WellKnownResolver
2021-09-27T15:54:47.4287856Z   File "/home/runner/work/synapse/synapse/synapse/http/federation/well_known_resolver.py", line 32, in <module>
2021-09-27T15:54:47.4288946Z     from synapse.util.caches.ttlcache import TTLCache
2021-09-27T15:54:47.4290319Z   File "/home/runner/work/synapse/synapse/synapse/util/caches/ttlcache.py", line 163, in <module>
2021-09-27T15:54:47.4291158Z     class _CacheEntry(Generic[KT, VT]):
2021-09-27T15:54:47.4309820Z   File "/home/runner/work/synapse/synapse/.tox/py/lib/python3.6/site-packages/attr/_make.py", line 1559, in wrap
2021-09-27T15:54:47.4310783Z     return builder.build_class()
2021-09-27T15:54:47.4313286Z   File "/home/runner/work/synapse/synapse/.tox/py/lib/python3.6/site-packages/attr/_make.py", line 730, in build_class
2021-09-27T15:54:47.4314208Z     return self._create_slots_class()
2021-09-27T15:54:47.4315773Z   File "/home/runner/work/synapse/synapse/.tox/py/lib/python3.6/site-packages/attr/_make.py", line 847, in _create_slots_class
2021-09-27T15:54:47.4316784Z     cls = type(self._cls)(self._cls.__name__, self._cls.__bases__, cd)
2021-09-27T15:54:47.4317678Z   File "/opt/hostedtoolcache/Python/3.6.15/x64/lib/python3.6/typing.py", line 948, in __new__
2021-09-27T15:54:47.4318610Z     raise TypeError("Cannot inherit from plain Generic")
2021-09-27T15:54:47.4319430Z builtins.TypeError: Cannot inherit from plain Generic
2021-09-27T15:54:47.4319918Z 
2021-09-27T15:54:47.4320688Z tests.rest.client.test_third_party_rules
```
</details>


Looks like https://github.com/python-attrs/attrs/issues/313